From a479c601016a69694625027d63e18f1106f7f92d Mon Sep 17 00:00:00 2001
From: Niklas Cassel <niklas.cassel@wdc.com>
Date: Tue, 17 Oct 2023 23:19:40 +0200
Subject: [PATCH 47/48] arm64: dts: rk3588-rock-5b: enable SATA for M.2 E-Key
 to rock-5b

We need to disable pcie2x1l0 in order to enable the onboard SATA
controller, as the top M.2 E-Key slot can be used by either the onboard
PCIe controller or the onboard SATA controller, but not both.

The M.2 E-Key slot is connected to a rockchip USB/PCIe/SATA combo PHY.
The SATA device tree node has a phandle to the combophy and supplies
the PHY mode which the PHY should be configured in (PHY_TYPE_SATA).

Signed-off-by: Niklas Cassel <niklas.cassel@wdc.com>
---
 arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts b/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
index a0e303c3a1dc..c5b779a0437d 100644
--- a/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts
@@ -137,7 +137,7 @@ &combphy1_ps {
 	status = "okay";
 };
 
-&combphy2_psu {
+&sata1 {
 	status = "okay";
 };
 
@@ -272,7 +272,7 @@ &pcie2x1l0 {
 	pinctrl-0 = <&pcie2_0_rst>;
 	reset-gpios = <&gpio4 RK_PA5 GPIO_ACTIVE_HIGH>;
 	vpcie3v3-supply = <&vcc3v3_pcie2x1l0>;
-	status = "okay";
+	status = "disabled";
 };
 
 &pcie2x1l2 {
-- 
2.44.0

